lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb

```diff
@@ -2,11 +2,20 @@
 require 'pathname'
 require 'cheffish'
 require 'chef_zero/socketless_server_map'
+require_relative 'ignore_convergence_failure'
 
 class Chef
 module Provisioning
   class ConvergenceStrategy
     class PrecreateChefObjects < ConvergenceStrategy
+
+      def initialize(convergence_options, config)
+        super
+        if convergence_options[:ignore_failure]
```

Comment:
I think you're working on elaborating this, but it seems like we want to be able to ignore audit failures, non-audit failures, or both.